### PR TITLE
[63] Querying search terms (keywords)

### LINKED ISSUE #63: Generating keyword searches
This issue should:

Produce a list of candidate GtR projects and Crunchbase organisations given a query of keywords

Inputs: 

- A set of keywords and key phrases (Note: these will be specific to CB and GtR datasets, and the preprocessing of the datasets has to be considered)

Outputs:

- A list of GtR project ids, and Crunchbase organisation ids pertaining to the cross-dataset categories of interest

### PR DESCRIPTION
**Pull request by beingkk**
Closes #63 

Implements simple keyword and key phrase search.

Usage examples are shown in `examples/06_query_terms.py`and in the [README](https://github.com/nestauk/innovation_sweet_spots/tree/63_keyword_searching/innovation_sweet_spots/analysis#querying).

NB: To run the notebook, you will need to first fetch updated, preprocessed data. Run the following from the terminal:
```
aws s3 sync s3://innovation-sweet-spots-lake/outputs/finals/pilot_outputs outputs/finals/pilot_outputs
```

---

Checklist:

- [x] I have refactored my code out from `notebooks/`
- [x] I have checked the code runs
- [x] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [x] I have merged any new changes from `dev`
- [x] I have documented the code
  - [x] Major functions have docstrings
  - [x] Appropriate information has been added to `README`s
- [x] I have explained the feature in this PR or (better) in `output/reports/`
- [x] I have requested a code review

### CODE REVIEW

**Comment by beingkk** [comment]
Hi @Jack-0-0 , hope it's OK for you to review this PR next week. Apologies, it's a bit heftier than usual.
The main additions are:
- module for keyword search [query_terms.py](https://github.com/nestauk/innovation_sweet_spots/pull/96/files#diff-6be4c522ebab36c810557816edf12a357979f41bc110d5a6e9189ef184a569cb)
- getter for preprocessed data [getters/preprocessed.py](https://github.com/nestauk/innovation_sweet_spots/pull/96/files#diff-9127034532cc941062d2857eb57f2a84df1dbdfcd12ff9348276302bad7649b1)

File: innovation_sweet_spots/analysis/query_terms.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            A dataframe with the following columns:
+                - a column for document identifiers
+                - a boolean column for each of the search queries, where True indicates a match
+                - a column 'has_any_terms' which indicates if any search queries where matched
+        """
+        matches = find_documents_with_set_of_terms(
+            search_queries, self.text_corpus, verbose=self.verbose
+        )
+        matches["id"] = list(self.document_ids)
+        if return_only_matches is True:
```
Comment (Jack-0-0) [inline innovation_sweet_spots/analysis/query_terms.py]: Can remove `is True` here.

**Review by Jack-0-0 — APPROVED** [review]
This PR is a great addition!
The notebook runs and works as expected. I like how the information is presented in the `find_matches` Query function.
The README looks good. The docstrings are clear throughout. The tests make sense and all pass.

File: innovation_sweet_spots/analysis/query_terms.py (hunk 1 above)
Comment (beingkk) [inline innovation_sweet_spots/analysis/query_terms.py]:
```suggestion
        if return_only_matches:
```

**Comment by beingkk** [comment]
Great, thanks so much @Jack-0-0 !

**Pull request merged**